fili-core/src/test/groovy/com/yahoo/bard/webservice/web/DataApiRequestSortSpec.groovy

```diff
@@ -0,0 +1,142 @@
+
+        then:
+        Exception e = thrown(BadApiRequestException)
+        e.getMessage() == expectedMessage
+    }
+
+    @Unroll
+    def "Validate the sort column and direction map from #sortString string"() {
+        when:
+        Map<String, SortDirection> actual = new DataApiRequest().generateSortColumns(sortString)
+
+        then:
+        actual == expected
+
+        where:
+        sortString                        | expected
+        "dateTime"                        | ["dateTime":SortDirection.DESC] as Map
+        "dateTime|ASC"                    | ["dateTime":SortDirection.ASC] as Map
+        "dateTimexyz|DESC"                | ["dateTimexyz":SortDirection.DESC] as Map
+        "xyz,dateTime,abc"                | ["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC, "abc":SortDirection.DESC] as Map
+        "xyz,dateTime|DESC,abc"           | ["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC, "abc":SortDirection.DESC] as Map
+        "xyz|DESC,dateTime|DESC,abc|ASC"  | ["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC, "abc":SortDirection.ASC] as Map
+        "xyz|DESC,dateTime,abc|DESC"      | ["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC, "abc":SortDirection.DESC] as Map
+        "xyz|DESC,dateTime"               | ["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC] as Map
+        ""                                | null
+        "dinga"                           | ["dinga":SortDirection.DESC] as Map
+
+    }
+
+    @Unroll
+    def "Generate dateTime sort column from columnDirection map "() {
+        when:
+        Optional<OrderByColumn> actual = new DataApiRequest().generateDateTimeSortColumn(columnDirection)
+
+        then:
+        actual == expected
+
+        where:
+        columnDirection                                                                          | expected
+        ["dateTime":SortDirection.DESC] as Map                                                   | Optional.of(new OrderByColumn("dateTime", SortDirection.DESC))
+        ["dateTime":SortDirection.ASC] as Map                                                    | Optional.of(new OrderByColumn("dateTime", SortDirection.ASC))
+        ["dateTimexyz":SortDirection.DESC] as Map                                                | Optional.empty()
+        ["dateTime":SortDirection.DESC, "abc":SortDirection.DESC] as Map                         | Optional.of(new OrderByColumn("dateTime", SortDirection.DESC))
+        null                                                                                     | Optional.empty()
+        ["dinga":SortDirection.DESC] as Map                                                      | Optional.empty()
+    }
+
+    @Unroll
+    def "Remove dateTime sort column from columnDirection map "() {
+        when:
+        Map<String, SortDirection> actual = new DataApiRequest().removeDateTimeSortColumn(columnDirection)
+
+        then:
+        actual == expected
```

Comment:
Can be simplified to just an `expected` block, since we're just comparing a value to a single call.